app/frontend/components/taskList/taskRow.tsx

```diff
@@ -0,0 +1,58 @@
+import { ErrorMessage } from '@hookform/error-message'
+import { yupResolver } from '@hookform/resolvers/yup'
+import { useForm } from 'react-hook-form'
+import { TaskFragment, useTaskUpdateMutation } from '../../generated/graphql'
+import { InputField } from '../inputField/inputField'
+import { CgSpinner } from 'react-icons/cg'
+import { TaskFormData, taskInputSchema } from './taskList'
+
+interface TaskRowProps {
+  task: TaskFragment
+}
+
+export const TaskRow = ({ task }: TaskRowProps) => {
+  const [{ fetching }, taskUpdate] = useTaskUpdateMutation()
+  const {
+    register,
+    handleSubmit,
+    formState: { errors },
+  } = useForm<TaskFormData>({
+    mode: 'onChange',
+    defaultValues: {
+      title: task.title,
+    },
+    resolver: yupResolver(taskInputSchema),
+  })
+
+  const handleSubmitTask = async (taskData: TaskFormData) => {
+    try {
+      const result = await taskUpdate({
+        id: task.id,
+        data: {
+          projectId: task.project.id,
+          title: taskData.title,
+        },
+      })
+      if (result.error) {
+        throw new Error(`GraphQL Error ${result.error}`)
+      }
+    } catch (error) {
+      // eslint-disable-next-line no-console
+      console.error(error)
```

Comment:
What will happen when the mutation fails?

Will the input have the original text again? Maybe it is worth to write a test for this edge case.

Reply:
i removed try catch